src/coprocessor/dag/aggr_fn/impl_max_min.rs

```diff
@@ -0,0 +1,339 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+use std::convert::TryFrom;
+
+use cop_codegen::AggrFunction;
+use cop_datatype::{EvalType, FieldTypeAccessor};
+use tipb::expression::{Expr, ExprType, FieldType};
+
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::mysql::Tz;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::dag::rpn_expr::{RpnExpression, RpnExpressionBuilder};
+use crate::coprocessor::Result;
+
+pub enum Extremum {
```

Comment:
I prefer to separate it for min and max, since this enum adds extra cost and we can actually avoid it.

Reply:
Maybe we can define an `Extremum` trait and make `AggrFnDefinitionParserExtremum` generic to reduce extra cost.

Reply:
addressed